api/src/migrations/main/1687288209602-transfer-operationYears-from-public-notice-to-projet.js

```diff
@@ -0,0 +1,52 @@
+const { MigrationInterface, QueryRunner } = require("typeorm");
+
+module.exports = class transferOperationYearsFromPublicNoticeToProjet1687288209602 {
+
+  async up(queryRunner) {
+		console.log('Starting project (add new columns - operation_start_year, operation_end_year) migration');
+		await queryRunner.query(`
+
+		-- add new column - operation_start_year, nullable (for existing data before new column exists)
+			ALTER TABLE app_fom.project ADD COLUMN operation_start_year integer;
+
+		-- add new column - operation_end_year, nullable (for existing data before new column exists)
+			ALTER TABLE app_fom.project ADD COLUMN operation_end_year integer;
+
+		-- comment on column - operation_start_year
+			COMMENT ON COLUMN app_fom.project.operation_start_year IS
+				'Starting Year of planned duration of the FOM operations';
+			
+		-- comment on column - operation_end_year
+			COMMENT ON COLUMN app_fom.project.operation_end_year IS
+				'Ending Year of planned duration of the FOM operations';
+
+		-- transfer data to new column - operation_start_year, nullable (for existing data before new column exists) from public_notice table
+			UPDATE app_fom.project
+			SET operation_start_year = public_notice.operation_start_year 
+			FROM app_fom.public_notice 
+			WHERE project.project_id = public_notice.project_id
+			AND public_notice.operation_start_year is not null;
+
+		-- transfer data to new column - operation_end_year, nullable (for existing data before new column exists) from public_notice table
+			UPDATE app_fom.project 
+			SET operation_end_year = public_notice.operation_end_year 
+			FROM app_fom.public_notice 
+			WHERE project.project_id = public_notice.project_id
+			AND public_notice.operation_end_year is not null;
+
```

Comment:
We don't drop the columns on public_notice, but we likely at least need to make them nullable.

Reply:
I'd be okay with dropping those columns too though.

Reply:
Maybe we could merge this pull request first, and verify if all the transfer data works in test deployment. And then I'll write another migration to drop those operation period columns in the public_notice table, and merge again. There are only 4 public notice in the prod database, and only one has the operation period data, should be safe to do so.

Reply:
I'm fine with that approach, that just means having a separate branch/PR with the 2nd migration to be merged afterward.